tests/test_algorithms.py

```diff
@@ -169,13 +169,45 @@ def test_from_ts(self):
             assert ts.sequence_length == 100
 
     def test_pedigree(self):
-        tables = simulate_pedigree(num_founders=10, num_generations=5)
+        num_founders = 10
+        num_generations = 5
+        tables = simulate_pedigree(
+            num_founders=num_founders, num_generations=num_generations
+        )
         with tempfile.TemporaryDirectory() as tmpdir:
             ts_path = pathlib.Path(tmpdir) / "pedigree.trees"
             tables.dump(ts_path)
             ts = self.run_script(f"0 --from-ts {ts_path} -r 0.1 --model=wf_ped")
         assert ts.num_trees > 1
         for tree in ts.trees():
             assert tree.num_roots > 1
-        assert ts.num_individuals == len(tables.individuals)
-        assert ts.num_samples == 20
+        for node in ts.nodes():
+            assert node.individual != tskit.NULL
+        assert (
+            len(tables.individuals)
+            == ts.num_individuals
+            == len(ts.dump_tables().individuals)
+        )
+        assert len(ts.dump_tables().nodes) == len(tables.nodes)
+        assert ts.num_samples == 2 * num_founders
+
+        # founder nodes have correct time
+        founder_ids = {
+            idx
+            for idx, ind in enumerate(ts.dump_tables().individuals)
+            if np.array_equal(ind.parents, [-1, -1])
+        }
+        assert all(
+            [
+                (node.individual in founder_ids)
+                for node in ts.dump_tables().nodes
+                if node.time == (num_generations - 1)
+            ]
+        )
+
```

Comment:
Good to make this a separate tests case, ``test_one_gen_pedigree`` or something.